packages/js/email-editor/src/components/preview/preview-save-guard.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback, useEffect } from '@wordpress/element';
+import { select, dispatch } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * A component that adds a confirmation dialog for unsaved changes
+ * when a user tries to open a preview in a new tab.
+ */
+export const PreviewSaveGuard = () => {
+	const selector = '.editor-preview-dropdown__button-external';
+
+	/**
+	 * Handles click/keydown events to check for unsaved changes before previewing.
+	 *
+	 * @param {Event} event The triggered event.
+	 */
+	const guard = async ( event ) => {
+		const target = event.target;
+
+		// Find the triggering element (the preview button).
+		const triggerEl = target?.closest( selector );
+		if ( ! triggerEl ) {
+			return;
+		}
+
+		// Check if there are any unsaved changes using the editor data store.
+		const editorStoreInstance = select( editorStore );
+		const isDirty = editorStoreInstance?.isEditedPostDirty?.();
```

Comment:
Could you explain to me why there is the last `.` here in the chain? I am not sure.

Reply:
The combination of `?.()` after `isEditedPostDirty` is a safety check. It ensures that the `isEditedPostDirty` method is only called if it exists on the `editorStoreInstance` object. If the method doesn't exist, the code simply stops and returns `undefined` instead of throwing a critical error.

Reply:
Right, makes sense, for me, this is quite an unusual syntax. And I am not sure if everyone else is familiar with this syntax, so I would do one of the following:

1. I don't think it is necessary; we don't have these safety checks in other places when we call methods on stores, so could we remove this?
2. Leave a comment explaining this and why it is necessary here. I would even replace it with the previous comment, which doesn't add much info. 
3. Use a different check to see if the method exists.